bootstrap_cfn/config.py

```diff
@@ -529,6 +534,54 @@ def ref_fixup(x):
                 return x
         return dict([(k, ref_fixup(v)) for k, v in o.items()])
 
+    def get_ec2_userdata(self):
+        data = self.data['ec2']
+
+        parts = []
+
+        boothook = self.get_hostname_boothook(data)
+
+        if boothook:
+            parts.append(boothook)
+
+        if "cloud_config" in data:
+            parts.append({
+                'content': yaml.dump(data['cloud_config']),
+                'mime_type': 'text/cloud-config'
+            })
+
+        if len(parts):
+            return mime_packer.pack(parts)
+
+    HOSTNAME_BOOTHOOK_TEMPLATE = textwrap.dedent("""\
+    #!/bin/sh
+    [ -e /etc/cloud/cloud.cfg.d/99_hostname.cfg ] || echo "hostname: {hostname}" > /etc/cloud/cloud.cfg.d/99_hostname.cfg
+    """)
+
+    def get_hostname_boothook(self, data):
+        if "hostname_pattern" not in data:
+            return None
```

Comment:
We could do I guess. We could default to 

- `{instance_id}.{tags[Role]}.{environment}.{application}`
  i.e. i-abcd123.docker.staging.pf; or
- `{instance_id}.{environment}.{application}`
  i.e. i-abcd123.staging.pf

Do you have a preference?